include/xtensor/xstrides.hpp

```diff
@@ -169,17 +171,57 @@ namespace xt
 
     namespace detail
     {
+        template <typename Arg, typename... Args>
+        struct last_type_helper {
+            using type = typename last_type_helper<Args...>::type;
+        };
+
+        template <typename Arg>
+        struct last_type_helper<Arg> {
+            using type = Arg;
+        };
+
+        template <typename... Args>
+        using last_type = typename last_type_helper<Args...>::type;
+
+        template <typename... Args>
+        constexpr bool last_type_is_missing = std::is_same<missing_type, last_type<Args...>>::value;
+
         template <std::size_t dim, class S>
         inline auto raw_data_offset(const S&) noexcept
         {
             using strides_value_type = std::decay_t<decltype(std::declval<S>()[0])>;
             return strides_value_type(0);
         }
 
+        template <std::size_t dim, class S>
+        inline auto raw_data_offset(const S& strides, missing_type) noexcept
+        {
+            using strides_value_type = std::decay_t<decltype(std::declval<S>()[0])>;
```

Comment:
I think `typename S::value_type` should be ok.

Reply:
I just copied what was there, but sounds good

Reply:
I can't really say ;) `std::decay_t<decltype(std::declval<S>()[0])>;` looks ugly ;) but it may severe someone. We can keep it in this PR and open a separate PR/issue on it?